barretenberg/cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield_impl.hpp

```diff
@@ -998,14 +1005,31 @@ bigfield<Builder, T> bigfield<Builder, T>::sqradd(const std::vector<bigfield>& t
     return remainder;
 }
 
-template <typename Builder, typename T> bigfield<Builder, T> bigfield<Builder, T>::pow(const size_t exponent) const
+template <typename Builder, typename T> bigfield<Builder, T> bigfield<Builder, T>::pow(const uint32_t exponent) const
 {
     // Just return one immediately
-
     if (exponent == 0) {
         return bigfield(uint256_t(1));
     }
 
+    // If this is a constant, compute result directly
+    if (is_constant()) {
+        auto base_val = get_value();
+        uint512_t result_val = 1;
+        uint512_t base = base_val % modulus_u512;
+        uint32_t shifted_exponent = exponent;
+
+        // Fast modular exponentiation
+        while (shifted_exponent > 0) {
+            if (shifted_exponent & 1) {
+                result_val = (uint1024_t(result_val) * uint1024_t(base) % uint1024_t(modulus_u512)).lo;
```

Comment:
Modular division is very slow in our code, so I'd recommend converting to a field element and then exponentiation

Reply:
Ok, talked about this. Not possible anymore, unfortunately, so we have to use your way